### ISSUE TITLE
AppData problem: tag-invalid : <release> version was duplicated

### ISSUE THREAD

**Issue by GreenCappuccino** [pobthebuilder/resolve-flatpak#7]
Running `flatpak-builder --force-clean --repo=repo build-dir com.blackmagic.Resolve.yaml` on Silverblue 38
```
Renaming com.blackmagic.Resolve.metainfo.xml to share/appdata/com.blackmagic.Resolve.appdata.xml
Running appstream-compose
Processing application com.blackmagic.Resolve
com.blackmagic.Resolve:            AppData problem: tag-invalid : <release> version was duplicated
Error loading AppData file: AppData file /app/share/appdata/com.blackmagic.Resolve.appdata.xml was not valid
Error: ERROR: appstream-compose failed: Child process exited with code 1
```

**Comment by GreenCappuccino** [comment]
Relevant section from generated appdata:
```xml
<?xml version="1.0" encoding="UTF-8"?>
[...]
  <releases>
    <release version="18.5.0" date="2023-05-04">
      <description>
This software installs DaVinci Resolve 18.5 public beta 2 which adds improvements to beta 1 release features including better quality when encoding AAC audio, improved consistency when decoding H.264 and H.265 clips on some Windows and AMD systems and more reliable syncing of Blackmagic Presentation markers. For Fusion, public beta 2 includes additional shading options in the Universal Scene Descriptor viewer and better control visibility options when working with uCamera. We’d like to thank everyone for the great feedback we have received. It’s been very helpful. Please continue to submit beta feedback on the Blackmagic Design forum.
      </description>
    </release>
    <release version="18.5.0" date="2023-04-17">
      <description>
This software update installs the DaVinci Resolve 18.5 public beta which adds support for new AI tools, as well as over 150 new features including new cut page editing toolset, new Fairlight audio mix automation and USD file support. Please submit all feedback and bug reports via the public beta feedback forum on the Blackmagic Design website.
      </description>
    </release>
[...]
  </releases>
</component>
```

**Comment by McCrafterIV** [comment]
This can be solved by replacing the version line in the appdata.sh file with the following:
```
"    <release version=\"" + (.urls["Linux"][0].major|tostring) + "." + (.urls["Linux"][0].minor|tostring) + "." + (.urls["Linux"][0].releaseNum|tostring) + "~" + (.numericDate|tostring) + "\" date=\"" + (.numericDate / 1000|strftime("%Y-%m-%d")) + "\">",
```
Adding the release date behind the version date solves the problem, but there is probably a better solution.

**Comment by pobthebuilder** [comment]
> This can be solved by replacing the version line in the appdata.sh file with the following:
> 
> ```
> "    <release version=\"" + (.urls["Linux"][0].major|tostring) + "." + (.urls["Linux"][0].minor|tostring) + "." + (.urls["Linux"][0].releaseNum|tostring) + "~" + (.numericDate|tostring) + "\" date=\"" + (.numericDate / 1000|strftime("%Y-%m-%d")) + "\">",
> ```
> 
> Adding the release date behind the version date solves the problem, but there is probably a better solution.

this seems to be the best fix for now. the beta version number is not encoded in the raw data so it would be hard to get at. i'll do this.

**Comment by pobthebuilder** [comment]
fixed by 2a3f364.
